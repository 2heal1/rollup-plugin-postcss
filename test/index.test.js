import path from 'path'
import fs from 'fs-extra'
import { rollup } from 'rollup'
import postcss from '../src'

function fixture(...args) {
  return path.join(__dirname, 'fixtures', ...args)
}

beforeAll(() => fs.remove(fixture('dist')))

async function write({
  input,
  dirname,
  options
}) {
  dirname = fixture('dist', dirname)
  const bundle = await rollup({
    input: fixture(input),
    plugins: [
      postcss(options)
    ]
  })
  await bundle.write({
    format: 'cjs',
    file: path.join(dirname, 'bundle.js')
  })
  const cssCodePath = path.join(dirname, 'bundle.css')
  const cssMapPath = path.join(dirname, 'bundle.css.map')
  const jsCodePath = path.join(dirname, 'bundle.js')
  return {
    jsCode() {
      return fs.readFile(jsCodePath, 'utf8')
    },
    cssCode() {
      return fs.readFile(cssCodePath, 'utf8')
    },
    cssMap() {
      return fs.readFile(cssMapPath, 'utf8')
    },
    hasCssFile() {
      return fs.pathExists(cssCodePath)
    },
    hasCssMapFile() {
      return fs.pathExists(cssMapPath)
    }
  }
}

function snapshot({
  title,
  input,
  options = {}
}) {
  test(title, async () => {
    const res = await write({
      input,
      dirname: title,
      options
    })

    expect(await res.jsCode()).toMatchSnapshot('js code')

    if (options.extract) {
      expect(await res.hasCssFile()).toBe(true)
      expect(await res.cssCode()).toMatchSnapshot('css code')
    }

    const sourceMap = options && options.sourceMap
    if (sourceMap === 'inline') {
      expect(await res.hasCssMapFile()).toBe(false)
    } else if (sourceMap === true) {
      expect(await res.hasCssMapFile()).toBe(Boolean(options.extract))
      if (options.extract) {
        expect(await res.cssMap()).toMatchSnapshot('css map')
      }
    }
  })
}

snapshot({
  title: 'simple',
  input: 'simple/index.js'
})

snapshot({
  title: 'extract',
  input: 'simple/index.js',
  options: {
    extract: true
  }
})

snapshot({
  title: 'minimize:inject',
  input: 'simple/index.js',
  options: {
    minimize: true
  }
})

snapshot({
  title: 'minimize:extract',
  input: 'simple/index.js',
  options: {
    minimize: true,
    extract: true
  }
})

snapshot({
  title: 'modules:inject',
  input: 'css-modules/index.js',
  options: {
    modules: true
  }
})

snapshot({
  title: 'modules:named-exports',
  input: 'named-exports/index.js',
  options: {
    modules: true,
    namedExports: true
  }
})

snapshot({
  title: 'modules:extract',
  input: 'css-modules/index.js',
  options: {
    modules: true,
    extract: true
  }
})

snapshot({
  title: 'sourcemap:true',
  input: 'simple/index.js',
  options: {
    sourceMap: true
  }
})

snapshot({
  title: 'extract::sourcemap:true',
  input: 'simple/index.js',
  options: {
    sourceMap: true,
    extract: true
  }
})

// inline is actually broken for now
snapshot({
  title: 'sourcemap:inline',
  input: 'simple/index.js',
  options: {
    sourceMap: 'inline'
  }
})

snapshot({
  title: 'extract::sourcemap:inline',
  input: 'simple/index.js',
  options: {
    sourceMap: 'inline',
    extract: true
  }
})

snapshot({
  title: 'inject:top',
  input: 'simple/index.js',
  options: {
    inject: {
      insertAt: 'top'
    }
  }
})

snapshot({
  title: 'sass',
  input: 'sass/index.js',
  options: {
    use: [
      'sass'
    ]
  }
})

snapshot({
  title: 'sass:sourcemap',
  input: 'sass/index.js',
  options: {
    sourceMap: true,
    use: [
      'sass'
    ]
  }
})

snapshot({
  title: 'postcss-config',
  input: 'postcss-config/index.js'
})

snapshot({
  title: 'sass:modules',
  input: 'sass-modules/index.js',
  options: {
    modules: true,
    use: [
      'sass'
    ]
  }
})

snapshot({
  title: 'skip-loader',
  input: 'simple/index.js',
  options: {
    use: ['loader'],
    loaders: [
      {
        name: 'loader',
        test: /\.random$/,
        process() {
          return 'lol'
        }
      }
    ]
  }
})

snapshot({
  title: 'postcss-options',
  input: 'postcss-options/index.js',
  options: {
    plugins: [
      require('autoprefixer')()
    ]
  }
})

snapshot({
  title: 'inject:false',
  input: 'simple/index.js',
  options: {
    inject: false
  }
})
