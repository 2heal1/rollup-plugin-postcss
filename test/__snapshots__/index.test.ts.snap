// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic postcss-config: js code 1`] = `
"'use strict';

function styleInject(css, ref) {
  if ( ref === void 0 ) ref = {};
  var insertAt = ref.insertAt;

  if (!css || typeof document === 'undefined') { return; }

  var head = document.head || document.getElementsByTagName('head')[0];
  var style = document.createElement('style');
  style.type = 'text/css';

  if (insertAt === 'top') {
    if (head.firstChild) {
      head.insertBefore(style, head.firstChild);
    } else {
      head.appendChild(style);
    }
  } else {
    head.appendChild(style);
  }

  if (style.styleSheet) {
    style.styleSheet.cssText = css;
  } else {
    style.appendChild(document.createTextNode(css));
  }
}

var css_248z = \\"a {\\\\r\\\\n  color: blue\\\\n}\\\\r\\\\n\\\\r\\\\n.multiline,\\\\r\\\\n.selector {\\\\r\\\\n  box-shadow: 1px 0 9px rgba(0, 0, 0, .4),\\\\r\\\\n              1px 0 3px rgba(0, 0, 0, .6)\\\\n}\\\\r\\\\n\\\\r\\\\n/*# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbInN0eWxlLnNzcyJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiQUFBQTtFQUNHO0FBREg7O0FBR0M7O0VBRUU7O0FBRE0iLCJmaWxlIjoic3R5bGUuc3NzIiwic291cmNlc0NvbnRlbnQiOlsiYVxyXG4gIGNvbG9yOiBibHVlXHJcblxyXG4ubXVsdGlsaW5lLFxyXG4uc2VsZWN0b3JcclxuICBib3gtc2hhZG93OiAxcHggMCA5cHggcmdiYSgwLCAwLCAwLCAuNCksXHJcbiAgICAgICAgICAgICAgMXB4IDAgM3B4IHJnYmEoMCwgMCwgMCwgLjYpXHJcbiJdfQ== */\\";
styleInject(css_248z);
"
`;

exports[`basic postcss-options: js code 1`] = `
"'use strict';

function styleInject(css, ref) {
  if ( ref === void 0 ) ref = {};
  var insertAt = ref.insertAt;

  if (!css || typeof document === 'undefined') { return; }

  var head = document.head || document.getElementsByTagName('head')[0];
  var style = document.createElement('style');
  style.type = 'text/css';

  if (insertAt === 'top') {
    if (head.firstChild) {
      head.insertBefore(style, head.firstChild);
    } else {
      head.appendChild(style);
    }
  } else {
    head.appendChild(style);
  }

  if (style.styleSheet) {
    style.styleSheet.cssText = css;
  } else {
    style.appendChild(document.createTextNode(css));
  }
}

var css_248z = \\"body {\\\\r\\\\n  display: -webkit-box;\\\\r\\\\n  display: -ms-flexbox;\\\\r\\\\n  display: flex;\\\\r\\\\n}\\\\r\\\\n\\\\r\\\\n/*# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbInN0eWxlLmNzcyJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiQUFBQTtFQUNFLG9CQUFhO0VBQWIsb0JBQWE7RUFBYixhQUFhO0FBQ2YiLCJmaWxlIjoic3R5bGUuY3NzIiwic291cmNlc0NvbnRlbnQiOlsiYm9keSB7XHJcbiAgZGlzcGxheTogZmxleDtcclxufVxyXG4iXX0= */\\";
styleInject(css_248z);
"
`;

exports[`basic skip-loader: js code 1`] = `
"'use strict';

function styleInject(css, ref) {
  if ( ref === void 0 ) ref = {};
  var insertAt = ref.insertAt;

  if (!css || typeof document === 'undefined') { return; }

  var head = document.head || document.getElementsByTagName('head')[0];
  var style = document.createElement('style');
  style.type = 'text/css';

  if (insertAt === 'top') {
    if (head.firstChild) {
      head.insertBefore(style, head.firstChild);
    } else {
      head.appendChild(style);
    }
  } else {
    head.appendChild(style);
  }

  if (style.styleSheet) {
    style.styleSheet.cssText = css;
  } else {
    style.appendChild(document.createTextNode(css));
  }
}

var css_248z = \\"body {\\\\r\\\\n  color: red;\\\\r\\\\n}\\\\r\\\\n\\\\r\\\\n/*# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbInN0eWxlLmNzcyJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiQUFBQTtFQUNFLFVBQVU7QUFDWiIsImZpbGUiOiJzdHlsZS5jc3MiLCJzb3VyY2VzQ29udGVudCI6WyJib2R5IHtcclxuICBjb2xvcjogcmVkO1xyXG59XHJcbiJdfQ== */\\";
styleInject(css_248z);
"
`;

exports[`modules auto-modules: js code 1`] = `
"'use strict';

function styleInject(css, ref) {
  if ( ref === void 0 ) ref = {};
  var insertAt = ref.insertAt;

  if (!css || typeof document === 'undefined') { return; }

  var head = document.head || document.getElementsByTagName('head')[0];
  var style = document.createElement('style');
  style.type = 'text/css';

  if (insertAt === 'top') {
    if (head.firstChild) {
      head.insertBefore(style, head.firstChild);
    } else {
      head.appendChild(style);
    }
  } else {
    head.appendChild(style);
  }

  if (style.styleSheet) {
    style.styleSheet.cssText = css;
  } else {
    style.appendChild(document.createTextNode(css));
  }
}

var css_248z = \\".a-module_foo {\\\\r\\\\n  color: red;\\\\r\\\\n}\\\\r\\\\n\\\\r\\\\n/*# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbImEubW9kdWxlLmNzcyJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiQUFBQTtFQUNFLFVBQVU7QUFDWiIsImZpbGUiOiJhLm1vZHVsZS5jc3MiLCJzb3VyY2VzQ29udGVudCI6WyIuZm9vIHtcclxuICBjb2xvcjogcmVkO1xyXG59XHJcbiJdfQ== */\\";
var a = {\\"foo\\":\\"a-module_foo\\"};
styleInject(css_248z);

var css_248z$1 = \\".b-module_b {\\\\r\\\\n  color: red;\\\\r\\\\n}\\\\r\\\\n\\\\r\\\\n/*# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbImIubW9kdWxlLnNjc3MiXSwibmFtZXMiOltdLCJtYXBwaW5ncyI6IkFBQUE7RUFDRSxVQUFVO0FBQ1oiLCJmaWxlIjoiYi5tb2R1bGUuc2NzcyIsInNvdXJjZXNDb250ZW50IjpbIi5iIHtcclxuICBjb2xvcjogcmVkO1xyXG59XHJcbiJdfQ== */\\";
var b = {\\"b\\":\\"b-module_b\\"};
styleInject(css_248z$1);

var css_248z$2 = \\".c-module_c {\\\\r\\\\n  color: red;\\\\r\\\\n}\\\\r\\\\n\\\\r\\\\n/*# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbImMubW9kdWxlLmxlc3MiXSwibmFtZXMiOltdLCJtYXBwaW5ncyI6IkFBQUE7RUFDRSxVQUFVO0FBQ1oiLCJmaWxlIjoiYy5tb2R1bGUubGVzcyIsInNvdXJjZXNDb250ZW50IjpbIi5jIHtcclxuICBjb2xvcjogcmVkO1xyXG59XHJcbiJdfQ== */\\";
var c = {\\"c\\":\\"c-module_c\\"};
styleInject(css_248z$2);

console.log(
  a,
  b,
  c
);
"
`;

exports[`modules extract: css code 1`] = `
"body {
  color: red;
}

.pcss {
  color: red;
}

.bar {
  color: red;
}

@nice-blue: #5B83AD;
@light-blue: @nice-blue + #111;

#header {
  color: @light-blue;
}

.style_foo {
  color: red;
}
"
`;

exports[`modules extract: js code 1`] = `
"'use strict';

var style = {\\"foo\\":\\"style_foo\\"};

console.log(style.foo);
"
`;

exports[`modules inject: js code 1`] = `
"'use strict';

function styleInject(css, ref) {
  if ( ref === void 0 ) ref = {};
  var insertAt = ref.insertAt;

  if (!css || typeof document === 'undefined') { return; }

  var head = document.head || document.getElementsByTagName('head')[0];
  var style = document.createElement('style');
  style.type = 'text/css';

  if (insertAt === 'top') {
    if (head.firstChild) {
      head.insertBefore(style, head.firstChild);
    } else {
      head.appendChild(style);
    }
  } else {
    head.appendChild(style);
  }

  if (style.styleSheet) {
    style.styleSheet.cssText = css;
  } else {
    style.appendChild(document.createTextNode(css));
  }
}

var css_248z = \\".style_foo {\\\\r\\\\n  color: red;\\\\r\\\\n}\\\\r\\\\n\\\\r\\\\n/*# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbInN0eWxlLmNzcyJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiQUFBQTtFQUNFLFVBQVU7QUFDWiIsImZpbGUiOiJzdHlsZS5jc3MiLCJzb3VyY2VzQ29udGVudCI6WyIuZm9vIHtcclxuICBjb2xvcjogcmVkO1xyXG59XHJcbiJdfQ== */\\";
var style = {\\"foo\\":\\"style_foo\\"};
styleInject(css_248z);

console.log(style.foo);
"
`;

exports[`modules inject-object: js code 1`] = `
"'use strict';

function styleInject(css, ref) {
  if ( ref === void 0 ) ref = {};
  var insertAt = ref.insertAt;

  if (!css || typeof document === 'undefined') { return; }

  var head = document.head || document.getElementsByTagName('head')[0];
  var style = document.createElement('style');
  style.type = 'text/css';

  if (insertAt === 'top') {
    if (head.firstChild) {
      head.insertBefore(style, head.firstChild);
    } else {
      head.appendChild(style);
    }
  } else {
    head.appendChild(style);
  }

  if (style.styleSheet) {
    style.styleSheet.cssText = css;
  } else {
    style.appendChild(document.createTextNode(css));
  }
}

var css_248z = \\".style_foo {\\\\r\\\\n  color: red;\\\\r\\\\n}\\\\r\\\\n\\\\r\\\\n/*# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbInN0eWxlLmNzcyJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiQUFBQTtFQUNFLFVBQVU7QUFDWiIsImZpbGUiOiJzdHlsZS5jc3MiLCJzb3VyY2VzQ29udGVudCI6WyIuZm9vIHtcclxuICBjb2xvcjogcmVkO1xyXG59XHJcbiJdfQ== */\\";
var style = {\\"foo\\":\\"style_foo\\"};
styleInject(css_248z);

console.log(style.foo);
"
`;

exports[`modules named-exports: js code 1`] = `
"'use strict';

function styleInject(css, ref) {
  if ( ref === void 0 ) ref = {};
  var insertAt = ref.insertAt;

  if (!css || typeof document === 'undefined') { return; }

  var head = document.head || document.getElementsByTagName('head')[0];
  var style = document.createElement('style');
  style.type = 'text/css';

  if (insertAt === 'top') {
    if (head.firstChild) {
      head.insertBefore(style, head.firstChild);
    } else {
      head.appendChild(style);
    }
  } else {
    head.appendChild(style);
  }

  if (style.styleSheet) {
    style.styleSheet.cssText = css;
  } else {
    style.appendChild(document.createTextNode(css));
  }
}

var foo = \\"style_foo\\";
var _new = \\"style_new\\";
var _default = \\"style_default\\";
var css_248z = \\".style_foo {\\\\n  color: red;\\\\n}\\\\n\\\\n.style_new {\\\\n  color: blue;\\\\n}\\\\n\\\\n.style_default {\\\\n  color: green;\\\\n}\\\\n\\\\n/*# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbInN0eWxlLmNzcyJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiQUFBQTtFQUNFLFVBQVU7QUFDWjs7QUFFQTtFQUNFLFdBQVc7QUFDYjs7QUFFQTtFQUNFLFlBQVk7QUFDZCIsImZpbGUiOiJzdHlsZS5jc3MiLCJzb3VyY2VzQ29udGVudCI6WyIuZm9vIHtcbiAgY29sb3I6IHJlZDtcbn1cblxuLm5ldyB7XG4gIGNvbG9yOiBibHVlO1xufVxuXG4uZGVmYXVsdCB7XG4gIGNvbG9yOiBncmVlbjtcbn1cbiJdfQ== */\\";
var style = {\\"foo\\":\\"style_foo\\",\\"new\\":\\"style_new\\",\\"default\\":\\"style_default\\",\\"_new\\":\\"style_new\\",\\"_default\\":\\"style_default\\"};
const stylesheet=\\".style_foo {\\\\n  color: red;\\\\n}\\\\n\\\\n.style_new {\\\\n  color: blue;\\\\n}\\\\n\\\\n.style_default {\\\\n  color: green;\\\\n}\\\\n\\\\n/*# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbInN0eWxlLmNzcyJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiQUFBQTtFQUNFLFVBQVU7QUFDWjs7QUFFQTtFQUNFLFdBQVc7QUFDYjs7QUFFQTtFQUNFLFlBQVk7QUFDZCIsImZpbGUiOiJzdHlsZS5jc3MiLCJzb3VyY2VzQ29udGVudCI6WyIuZm9vIHtcbiAgY29sb3I6IHJlZDtcbn1cblxuLm5ldyB7XG4gIGNvbG9yOiBibHVlO1xufVxuXG4uZGVmYXVsdCB7XG4gIGNvbG9yOiBncmVlbjtcbn1cbiJdfQ== */\\";
styleInject(css_248z);

var style$1 = /*#__PURE__*/Object.freeze({
  __proto__: null,
  foo: foo,
  _new: _new,
  _default: _default,
  'default': style,
  stylesheet: stylesheet
});

console.log(style$1);
"
`;

exports[`modules named-exports-custom-class-name: js code 1`] = `
"'use strict';

function styleInject(css, ref) {
  if ( ref === void 0 ) ref = {};
  var insertAt = ref.insertAt;

  if (!css || typeof document === 'undefined') { return; }

  var head = document.head || document.getElementsByTagName('head')[0];
  var style = document.createElement('style');
  style.type = 'text/css';

  if (insertAt === 'top') {
    if (head.firstChild) {
      head.insertBefore(style, head.firstChild);
    } else {
      head.appendChild(style);
    }
  } else {
    head.appendChild(style);
  }

  if (style.styleSheet) {
    style.styleSheet.cssText = css;
  } else {
    style.appendChild(document.createTextNode(css));
  }
}

var foohacked = \\"style_foo\\";
var newhacked = \\"style_new\\";
var defaulthacked = \\"style_default\\";
var css_248z = \\".style_foo {\\\\n  color: red;\\\\n}\\\\n\\\\n.style_new {\\\\n  color: blue;\\\\n}\\\\n\\\\n.style_default {\\\\n  color: green;\\\\n}\\\\n\\\\n/*# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbInN0eWxlLmNzcyJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiQUFBQTtFQUNFLFVBQVU7QUFDWjs7QUFFQTtFQUNFLFdBQVc7QUFDYjs7QUFFQTtFQUNFLFlBQVk7QUFDZCIsImZpbGUiOiJzdHlsZS5jc3MiLCJzb3VyY2VzQ29udGVudCI6WyIuZm9vIHtcbiAgY29sb3I6IHJlZDtcbn1cblxuLm5ldyB7XG4gIGNvbG9yOiBibHVlO1xufVxuXG4uZGVmYXVsdCB7XG4gIGNvbG9yOiBncmVlbjtcbn1cbiJdfQ== */\\";
var style = {\\"foo\\":\\"style_foo\\",\\"new\\":\\"style_new\\",\\"default\\":\\"style_default\\",\\"foohacked\\":\\"style_foo\\",\\"newhacked\\":\\"style_new\\",\\"defaulthacked\\":\\"style_default\\"};
const stylesheet=\\".style_foo {\\\\n  color: red;\\\\n}\\\\n\\\\n.style_new {\\\\n  color: blue;\\\\n}\\\\n\\\\n.style_default {\\\\n  color: green;\\\\n}\\\\n\\\\n/*# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbInN0eWxlLmNzcyJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiQUFBQTtFQUNFLFVBQVU7QUFDWjs7QUFFQTtFQUNFLFdBQVc7QUFDYjs7QUFFQTtFQUNFLFlBQVk7QUFDZCIsImZpbGUiOiJzdHlsZS5jc3MiLCJzb3VyY2VzQ29udGVudCI6WyIuZm9vIHtcbiAgY29sb3I6IHJlZDtcbn1cblxuLm5ldyB7XG4gIGNvbG9yOiBibHVlO1xufVxuXG4uZGVmYXVsdCB7XG4gIGNvbG9yOiBncmVlbjtcbn1cbiJdfQ== */\\";
styleInject(css_248z);

var style$1 = /*#__PURE__*/Object.freeze({
  __proto__: null,
  foohacked: foohacked,
  newhacked: newhacked,
  defaulthacked: defaulthacked,
  'default': style,
  stylesheet: stylesheet
});

console.log(style$1);
"
`;

exports[`onExtract 1`] = `
"'use strict';

console.log(undefined, undefined);
"
`;

exports[`sass data-prepend: js code 1`] = `
"'use strict';

function styleInject(css, ref) {
  if ( ref === void 0 ) ref = {};
  var insertAt = ref.insertAt;

  if (!css || typeof document === 'undefined') { return; }

  var head = document.head || document.getElementsByTagName('head')[0];
  var style = document.createElement('style');
  style.type = 'text/css';

  if (insertAt === 'top') {
    if (head.firstChild) {
      head.insertBefore(style, head.firstChild);
    } else {
      head.appendChild(style);
    }
  } else {
    head.appendChild(style);
  }

  if (style.styleSheet) {
    style.styleSheet.cssText = css;
  } else {
    style.appendChild(document.createTextNode(css));
  }
}

var css_248z = \\".special {\\\\n  color: $special-color;\\\\n}\\\\n\\\\n/*# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbInN0eWxlLnNjc3MiXSwibmFtZXMiOltdLCJtYXBwaW5ncyI6IkFBQUE7RUFDRSxxQkFBcUI7QUFDdkIiLCJmaWxlIjoic3R5bGUuc2NzcyIsInNvdXJjZXNDb250ZW50IjpbIi5zcGVjaWFsIHtcbiAgY29sb3I6ICRzcGVjaWFsLWNvbG9yO1xufVxuIl19 */\\";
styleInject(css_248z);
"
`;

exports[`sass data-prepend: js code 2`] = `
"'use strict';

function styleInject(css, ref) {
  if ( ref === void 0 ) ref = {};
  var insertAt = ref.insertAt;

  if (!css || typeof document === 'undefined') { return; }

  var head = document.head || document.getElementsByTagName('head')[0];
  var style = document.createElement('style');
  style.type = 'text/css';

  if (insertAt === 'top') {
    if (head.firstChild) {
      head.insertBefore(style, head.firstChild);
    } else {
      head.appendChild(style);
    }
  } else {
    head.appendChild(style);
  }

  if (style.styleSheet) {
    style.styleSheet.cssText = css;
  } else {
    style.appendChild(document.createTextNode(css));
  }
}

var css_248z = \\".special {\\\\n  color: $special-color;\\\\n}\\\\n\\\\n/*# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbInN0eWxlLnNjc3MiXSwibmFtZXMiOltdLCJtYXBwaW5ncyI6IkFBQUE7RUFDRSxxQkFBcUI7QUFDdkIiLCJmaWxlIjoic3R5bGUuc2NzcyIsInNvdXJjZXNDb250ZW50IjpbIi5zcGVjaWFsIHtcbiAgY29sb3I6ICRzcGVjaWFsLWNvbG9yO1xufVxuIl19 */\\";
styleInject(css_248z);
"
`;

exports[`sass import: js code 1`] = `
"'use strict';

function styleInject(css, ref) {
  if ( ref === void 0 ) ref = {};
  var insertAt = ref.insertAt;

  if (!css || typeof document === 'undefined') { return; }

  var head = document.head || document.getElementsByTagName('head')[0];
  var style = document.createElement('style');
  style.type = 'text/css';

  if (insertAt === 'top') {
    if (head.firstChild) {
      head.insertBefore(style, head.firstChild);
    } else {
      head.appendChild(style);
    }
  } else {
    head.appendChild(style);
  }

  if (style.styleSheet) {
    style.styleSheet.cssText = css;
  } else {
    style.appendChild(document.createTextNode(css));
  }
}

var css_248z = \\"@import \\\\\\"~foo/bar/a\\\\\\";\\\\r\\\\n@import \\\\\\"~foo/bar/b\\\\\\";\\\\r\\\\n@import \\\\\\"~foo/bar/c\\\\\\";\\\\r\\\\n@import \\\\\\"~foo/bar/partial\\\\\\";\\\\r\\\\n@import \\\\\\"foo.scss\\\\\\";\\\\r\\\\n\\\\r\\\\n/*# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbInN0eWxlLnNjc3MiXSwibmFtZXMiOltdLCJtYXBwaW5ncyI6IkFBQUEsb0JBQW9CO0FBQ3BCLG9CQUFvQjtBQUNwQixvQkFBb0I7QUFDcEIsMEJBQTBCO0FBQzFCLGtCQUFrQiIsImZpbGUiOiJzdHlsZS5zY3NzIiwic291cmNlc0NvbnRlbnQiOlsiQGltcG9ydCBcIn5mb28vYmFyL2FcIjtcclxuQGltcG9ydCBcIn5mb28vYmFyL2JcIjtcclxuQGltcG9ydCBcIn5mb28vYmFyL2NcIjtcclxuQGltcG9ydCBcIn5mb28vYmFyL3BhcnRpYWxcIjtcclxuQGltcG9ydCBcImZvby5zY3NzXCI7XHJcbiJdfQ== */\\";
styleInject(css_248z);
"
`;
